libs/ledger/include/ledger/chain/main_chain.hpp

```diff
@@ -216,6 +213,10 @@ class MainChain
   BlockMap::size_type UncacheBlock(BlockHash const &hash) const;
   void                KeepBlock(IntBlockPtr const &block) const;
   bool LoadBlock(BlockHash const &hash, Block &block, BlockHash *next_hash = nullptr) const;
+  template <class ParentHash, class ChildHash>
```

Comment:
I wouldn't make these templates unless absolutely neccessary, which I don't think they are. Could I suggest ```void CacheReference(BlockHash const &parent...)```

Reply:
> Could I suggest void CacheReference(BlockHash const &parent...)
Sure. The original implementation was templated to acommodate for various [lr]value combinations but as it turns out this method is called from a single site as of now and they are pretty rvalue there.